core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -93,6 +93,14 @@ class SparkEnv (
     // actorSystem.awaitTermination()
 
     // Note that blockTransferService is stopped by BlockManager since it is started by it.
+    
+    // If we only stop sc, but the driver process still run as a services then we need to delete 
+    // the tmp dir, if not, it will create too many tmp dirs
+    try {
+      Utils.deleteRecursively(new File(sparkFilesDir))
```

Comment:
I'm not sure this one is safe. Looking at the code, `sparkFilesDir` will be the current working directory on executors. Is that really safe to delete recursively?